src/test/java/htsjdk/samtools/CRAMAllEncodingStrategiesTest.java

```diff
@@ -21,32 +21,72 @@ public class CRAMAllEncodingStrategiesTest extends HtsjdkTest {
+                { new File(TEST_DATA_DIR, "mitoAlignmentStartTestGATKGen.cram"),
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"), true, false },
+                { new File(TEST_DATA_DIR, "mitoAlignmentStartTest.cram"),
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"), true, false },
+                // files created by rewriting the htsjdk test file src/test/resources/htsjdk/samtools/cram/mitoAlignmentStartTest.cram
+                // using code that replicates the first read (which is aligned to position 1 of the mito contig) either
+                // 10,000 or 20,000 times, to create a file with 2 or 3 containers, respectively, that have reads aligned to
+                // position 1 of the contig
+                { new File(TEST_DATA_DIR, "mitoAlignmentStartTest_2_containers_aligned_to_pos_1.cram"),
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"), true, false },
+                { new File(TEST_DATA_DIR, "mitoAlignmentStartTest_3_containers_aligned_to_pos_1.cram"),
+                        new File(TEST_DATA_DIR, "mitoAlignmentStartTest.fa"), true, false }
         };
     }
 
-    @Test(dataProvider = "roundTripTestFiles")
-    public final void testRoundTripDefaultEncodingStrategy(final File sourceFile, final File referenceFile) throws IOException {
+    @Test(dataProvider = "defaultStrategyRoundTripTestFiles")
+    public final void testRoundTripDefaultEncodingStrategy(
+            final File sourceFile,
+            final File referenceFile,
+            final boolean lenientEquality,
+            final boolean emitDetail) throws IOException {
         final CRAMEncodingStrategy testStrategy = new CRAMEncodingStrategy();
         final File tempOutCRAM = File.createTempFile("testRoundTrip", ".cram");
         tempOutCRAM.deleteOnExit();
         CRAMTestUtils.writeToCRAMWithEncodingStrategy(testStrategy, sourceFile, tempOutCRAM, referenceFile);
-        assertRoundTripFidelity(sourceFile, tempOutCRAM, referenceFile, false);
-        assertRoundtripFidelityWithSamtools(tempOutCRAM, referenceFile);
+        assertRoundTripFidelity(sourceFile, tempOutCRAM, referenceFile, lenientEquality, emitDetail);
+        assertRoundtripFidelityWithSamtools(tempOutCRAM, referenceFile, lenientEquality, emitDetail);
+    }
+
+    @DataProvider(name="encodingStrategiesTestFiles")
+    public Object[][] encodingStrategiesTestFiles() {
+        return new Object[][] {
+                { new File(TEST_DATA_DIR, "NA12878.20.21.1-100.100-SeqsPerSlice.500-unMapped.cram"),
+                        new File(TEST_DATA_DIR, "human_g1k_v37.20.21.1-100.fasta"), false, false },
+        };
     }
 
-    @Test(dataProvider = "roundTripTestFiles")
-    public final void testAllEncodingStrategyCombinations(final File cramSourceFile, final File referenceFile) throws IOException {
+    @Test(dataProvider = "encodingStrategiesTestFiles")
```

Comment:
Would there be value in having this `testAllEncodingStrategyCombinations ` test use the `defaultStrategyRoundTripTestFiles` DataProvider as well, with its many additional test cases?

Reply:
Perhaps, but there are currently 81 encoding strategies, so using all of those files would make this test take a really long time, and I would guess, dominate the CI test time, especially now that we have the large CRAM test file in that list. So instead I separated them this way.